pkg/kinflate/resource/configmap_test.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	manifest "k8s.io/kubectl/pkg/apis/manifest/v1alpha1"
+	"k8s.io/kubectl/pkg/loader"
+)
+
+// TODO: move it to the loader pkg ?
+type fakeLoader struct {
+	content string
+}
+
+func (f *fakeLoader) New(root string) (loader.Loader, error) {
+	return f, nil
+}
+
+func (f *fakeLoader) Load(location string) ([]byte, error) {
+	return []byte(f.content), nil
+}
+
+func makeEnvConfigMap(name string) *corev1.ConfigMap {
```

Comment:
Note for later: I realize this is how the tests have been written initially. But I usually keep the tests as simple as possible (even if it means copy-pasting more) by having the explicit input and expectations within the test.